src/FwdState.cc

```diff
@@ -387,19 +386,8 @@ FwdState::Start(const Comm::ConnectionPointer &clientConn, StoreEntry *entry, Ht
         return;
     }
 
-    switch (request->url.getScheme()) {
-
-    case AnyP::PROTO_URN:
-        urnStart(request, entry, al);
-        return;
-
-    default:
-        FwdState::Pointer fwd = new FwdState(clientConn, entry, request, al);
-        fwd->start(fwd);
-        return;
-    }
-
-    /* NOTREACHED */
+    FwdState::Pointer fwd = new FwdState(clientConn, entry, request, al);
+    fwd->start(fwd);
 }
```

Comment:
[AFAIK](https://github.com/squid-cache/squid/pull/915#pullrequestreview-873988160), official Squid does not forward requests with AnyP::PROTO_URN URIs to peers. _This_ PR should not _start_ doing that. Correctly forwarding such requests is a serious change that affects a lot of code that has not seen such URIs (not to mention potentially vulnerable cache_peers that have not seen them!). Introducing such new risks is probably a bad idea overall and is definitely outside this PR scope.

I see at least two ways to properly address this concern:

1: Respond with an error if a AnyP::PROTO_URN request gets to FwdState::Start(), where official code calls urnStart(). This is a minimal change. We can even preserve urn.cc (with nothing but an error-generating urnStart()) to minimize changes introduced by this PR.

2: Drop all AnyP::PROTO_URN code. This change would greatly simplify several code areas and address concerns about existing URN parsing code quality (that this PR does not address).

Given the urgency, my recommendation is to implement item 1 in this PR and item 2 in a non-rushed followup PR. Please let me know if you want me to do that.

Reply:
@rousskov , URNs are very much still a thing. All that has changed is deprecation of the particular "Trivial HTTP (RFC2169)" mechanism Squid implemented. All HTTP agents are still expected to be able to handle URNs in a appropriate manner - for Squid that is at least relaying to some upstream agent (*without* using Trivial-HTTP now).

What the peer/origin does with the URN is no longer relevant to Squid. If rejection is called for then it is *that* agents decision, not ours ... unless the squid.conf specifically says to deny based on a"proto URN" ACL.

Reply:
To be even clearer: scope of this PR is to *change* the mechanism for relay upstream from Trivial-HTTP to native HTTP.

Reply:
Please keep the scope of this PR limited to dropping support, especially when [using](https://github.com/squid-cache/squid/pull/915#issuecomment-2111245689) arguments like "urgent matter for security vulnerabilities" in PR discussions. Drop the problematic code first; that can be done in a matter of hours -- this PR is almost ready. After this PR lands, add URN request forwarding functionality in a separate not-rushed, carefully-written, and well-tested PR.